inspirehep/modules/authors/utils.py

```diff
@@ -126,3 +137,79 @@ def phonetic_blocks(full_names, phonetic_algorithm='nysiis'):
     )
 
     return dict(zip(full_names, phonetic_blocks))
+
+
+def formdata_to_model(formdata, id_workflow, id_user, is_update=False):
+    """Generate the record model dict and extra_data dict from the form data.
+
+    Returns:
+        tuple(dict, dict): tuple containing the dict representaion of a record
+            and the extra_data to create a new workflow.
+    """
+    form_fields = copy.deepcopy(formdata)
+    extra_data = {}
+
+    filter_empty_elements(
+        form_fields,
+        ['institution_history', 'advisors',
+         'websites', 'experiments']
+    )
+    data = updateform.do(form_fields)
+
+    data['_collections'] = ['Authors']
+
+    if '$schema' in data and not data['$schema'].startswith('http'):
+        data['$schema'] = url_for(
+            'invenio_jsonschemas.get_schema',
+            schema_path="records/{0}".format(data['$schema'])
+        )
+
+    author_name = ''
+
+    if 'family_name' in form_fields and form_fields['family_name']:
+        author_name = form_fields['family_name'].strip() + ', '
+    if 'given_names' in form_fields and form_fields['given_names']:
+        author_name += form_fields['given_names']
+
+    if author_name:
+        data.get('name', {})['value'] = author_name
+
+    if 'extra_comments' in form_fields and form_fields['extra_comments']:
+        data.setdefault('_private_notes', []).append({
+            'source': 'submitter',
+            'value': form_fields['extra_comments']
+        })
+
+    data['stub'] = False
+
+    # ==========
+    # Submitter Info
+    # ==========
+    try:
+        user_email = User.query.get(id_user).email
+    except AttributeError:
+        user_email = ''
+    try:
+        orcid = UserIdentity.query.filter_by(
+            id_user=id_user,
+            method='orcid'
+        ).one().id
+    except NoResultFound:
+        orcid = ''
+    data['acquisition_source'] = dict(
+        email=user_email,
+        datetime=datetime.datetime.utcnow().isoformat(),
+        method="submitter",
+        orcid=orcid,
+        submission_number=str(id_workflow),
+        internal_uid=int(id_user),
+    )
+
+    data = strip_empty_values(data)
+
+    validate(data, 'authors')
```

Comment:
Maybe validation won't be needed here in the future if it will happen at the beginning of a workflow the same it is now happening for hep records. This way, even if the data is not valid, it will not be lost, but saved in an ERRORed workflow so as to make the problem visible and, why not, easy to fix.

Reply:
Yep, though we don't have validation on the authors workflow yet, so it should probably still be there until we tackle those (in any case I think it should be low prio, as the data that the forms send should always be valid, as it's not something that is populated by external sources like the hepcrawl does).

Reply:
> in any case I think it should be low prio, as the data that the forms send should always be valid

50 users in the past three months have been victim of this assumption: https://sentry.inspirehep.net/inspire-sentry/prod/issues/48/